packages/itwinui-react/src/react-table/react-table.ts

```diff
@@ -0,0 +1,1323 @@
+  isSorted: boolean;
+  sortedIndex: number;
+  isSortedDesc: boolean | undefined;
+}
+
+export type OrderByFn<D extends Record<string, unknown>> = (
+  rowA: Row<D>,
+  rowB: Row<D>,
+) => number;
+export type SortByFn<D extends Record<string, unknown>> = (
+  rowA: Row<D>,
+  rowB: Row<D>,
+  columnId: IdType<D>,
+  desc?: boolean,
+) => number;
+
+export type DefaultSortTypes =
+  | 'alphanumeric'
+  | 'datetime'
+  | 'basic'
+  | 'string'
+  | 'number';
+
+export interface SortingRule<D> {
+  id: IdType<D>;
+  desc?: boolean | undefined;
+}
+
+//#endregion
+
+// Additional API
+export declare const actions: Record<string, string>;
+export type ActionType = { type: string } & Record<string, any>;
+export declare const defaultColumn: Partial<Column> & Record<string, any>;
+
+// Helpers
+export type StringKey<D> = Extract<keyof D, string>;
+export type IdType<D> = StringKey<D> | string;
+export type CellValue<V = any> = V;
+
+export type Renderer<Props> =
+  | ComponentType<Props>
+  | ReactElement
+  | string
+  | number
+  | ReactFragment;
+
+export interface PluginHook<D extends Record<string, unknown>> {
+  (hooks: Hooks<D>): void;
+  pluginName?: string | undefined;
+}
+
+export type TableDispatch<A = any> = (action: A) => void;
+
+// utils
+export declare function defaultOrderByFn<
+  D extends Record<string, unknown> = {},
+>(
+  arr: Array<Row<D>>,
+  funcs: Array<OrderByFn<D>>,
+  dirs: boolean[],
+): Array<Row<D>>;
+
+export declare function defaultGroupByFn<
+  D extends Record<string, unknown> = {},
+>(rows: Array<Row<D>>, columnId: IdType<D>): Record<string, Array<Row<D>>>;
+
+export declare function makePropGetter(hooks: Hooks, ...meta: any[]): any;
+
+export declare function reduceHooks<T extends object = {}>(
+  hooks: Hooks,
+  initial: T,
+  ...args: any[]
+): T;
+
+export declare function loopHooks(hooks: Hooks, ...args: any[]): void;
+
+export declare function ensurePluginOrder<
+  D extends Record<string, unknown> = {},
+>(plugins: Array<PluginHook<D>>, befores: string[], pluginName: string): void;
+
+export declare function functionalUpdate<
+  D extends Record<string, unknown> = {},
+>(updater: any, old: Partial<TableState<D>>): Partial<TableState<D>>;
+
+export declare function useGetLatest<T>(obj: T): () => T;
+
+export declare function safeUseLayoutEffect(
+  effect: EffectCallback,
+  deps?: DependencyList,
+): void;
+
+export declare function useMountedLayoutEffect(
+  effect: EffectCallback,
+  deps?: DependencyList,
+): void;
+
+export declare function useAsyncDebounce<F extends (...args: any[]) => any>(
+  defaultFn: F,
+  defaultWait?: number,
+): F;
+
+export declare function makeRenderer(
+  instance: TableInstance,
+  column: ColumnInstance,
+  meta?: any,
+): ReactElement;
+
+//-------------------------------------------------------------------------------
+// Custom Additions (parts from the old file called react-table-config.ts)
+
+export type FieldType = 'text' | 'number' | 'date' | string;
```

Comment:
is it possible to move these custom additions to the top of the file? this way we can be sure everything after it is a fork of react-table

Reply:
https://github.com/iTwin/iTwinUI/pull/1514/commits/ac68f3f540ea6516ecf1536861857705d402df68: Sounds good, moved.

Reply:
the `TableFilterProps` can also go in the custom additions section

also i think the `import`s need to be at the very top

Reply:
> the `TableFilterProps` can also go in the custom additions section

https://github.com/iTwin/iTwinUI/pull/1514/commits/cf815ff38bbb4953c0486f9924369e1518b673d3: Ohh, right, done.

> also i think the `import`s need to be at the very top

I thought of including the imports in the forked section of the file since the imports were originally a part of the `@types/react-table` file. But yes, putting imports at the top is more correct in case if custom additions also need to import. So I made this change (https://github.com/iTwin/iTwinUI/pull/1514/commits/cf815ff38bbb4953c0486f9924369e1518b673d3).